src/services/disable-method.js

```diff
@@ -0,0 +1,23 @@
+
+import feathersErrors from 'feathers-errors';
+
+const errors = feathersErrors.errors;
+
+export default function (...providers) {
+  return hook => {
+    const hookProvider = (hook.params || {}).provider;
+
+    const anyProvider = providers.length === 0;
+    const thisProvider = providers.some(provider =>
+      provider === hookProvider ||
+      (provider === 'server' && !hookProvider) ||

```

Comment:
The correct provider settings are:

- external: `!!provider`
- REST: `provider === 'rest'`
- Socket.io: `provider === 'socketio'`
- Primus: `provider === 'primus'`